modules/kafka/src/main/java/org/testcontainers/containers/KafkaContainer.java

```diff
@@ -182,15 +182,14 @@ protected void containerIsStarting(InspectContainerResponse containerInfo) {
         // exporting KAFKA_ADVERTISED_LISTENERS with the container hostname
         command += String.format("export KAFKA_ADVERTISED_LISTENERS=%s\n", kafkaAdvertisedListeners);
 
-        if (this.kraftEnabled && isLessThanCP740()) {
+        if (!this.kraftEnabled || isLessThanCP740()) {
             // Optimization: skip the checks
             command += "echo '' > /etc/confluent/docker/ensure \n";
-            command += commandKraft();
         }
 
-        if (!this.kraftEnabled) {
```

Comment:
why not just add `if (!this.kraftEnabled && this.externalZookeeperConnect == null) {`.

Reply:
We still need the 
```
command += "echo '' > /etc/confluent/docker/ensure \n";
```
even when not in kraft or external zookeeper mode.

Reply:
instead of all the changes in this PR, I just made the one I mentioned and it was working as expected.

Reply:
Could you give the full method snippet? I am not sure I follow. You can skip the following line, but as was mentioned in the original code, it is an optimization and everything should still work without it, but probably slower.
```
command += "echo '' > /etc/confluent/docker/ensure \n";
```

Reply:
the whole change I did is updating the if statement

Reply:
Then for the case where it's not Kraft mode (`!this.kraftEnabled`) but the external zookeeper is used (`this.externalZookeeperConnect != null`) we will miss the following line in our command
```java
command += "echo '' > /etc/confluent/docker/ensure \n";
```
which is supposed to give some optimization in boot time, as I was saying previously.